swatch-tally/ct/java/utils/TallyTestHelpers.java

```diff
@@ -0,0 +1,199 @@
+
+    OffsetDateTime timestamp = OffsetDateTime.parse(timestampStr);
+    OffsetDateTime expiration = timestamp.plusDays(EVENT_EXPIRATION_DAYS);
+
+    Event event = new Event();
+    event.setEventId(UUID.fromString(eventIdStr));
+    event.setOrgId(orgId);
+    event.setInstanceId(instanceId);
+    event.setDisplayName(Optional.of("Test Instance"));
+    event.setTimestamp(timestamp);
+    event.setRecordDate(timestamp);
+    event.setExpiration(Optional.of(expiration));
+    event.setEventSource("cost-management");
+    event.setEventType("snapshot");
+    event.setSla(Event.Sla.PREMIUM);
+    event.setRole(Event.Role.RED_HAT_ENTERPRISE_LINUX_SERVER);
+    event.setUsage(Event.Usage.PRODUCTION);
+    event.setServiceType("RHEL System");
+    event.setHardwareType(Event.HardwareType.CLOUD);
+    event.setCloudProvider(Event.CloudProvider.AWS);
+    event.setBillingProvider(Event.BillingProvider.AWS);
+    event.setBillingAccountId(Optional.of(DEFAULT_BILLING_ACCOUNT));
+    event.setConversion(true);
+    event.setProductIds(List.of(TEST_PRODUCT_ID));
+    event.setProductTag(Set.of(DEFAULT_PRODUCT_TAG));
+
+    var measurement = new Measurement();
+    measurement.setValue((double) value);
+    measurement.setMetricId("VCPUS");
+    event.setMeasurements(List.of(measurement));
+
+    return event;
+  }
+
+  public void syncTallyNightly(String orgId, SwatchService service) throws Exception {
+    Response response =
+        service
+            .given()
+            .header("x-rh-swatch-psk", TEST_PSK)
+            .header("x-rh-swatch-synchronous-request", "true")
+            .put("/api/rhsm-subscriptions/v1/internal/rpc/tally/snapshots/" + orgId)
+            .then()
+            .extract()
+            .response();
+
+    if (response.getStatusCode() != 200) {
+      throw new RuntimeException(
+          "Tally sync failed with status code: "
+              + response.getStatusCode()
+              + ", response body: "
+              + response.getBody().asString());
+    }
+
+    Log.info("Sync nightly tally endpoint called successfully for org: %s", orgId);
+  }
+
+  public void syncTallyHourly(String orgId, SwatchService service) throws Exception {
+    Response response =
+        service
+            .given()
+            .header("x-rh-swatch-psk", TEST_PSK)
+            .queryParam("org", orgId)
+            .post("/api/rhsm-subscriptions/v1/internal/tally/hourly")
+            .then()
+            .extract()
+            .response();
+
+    if (response.getStatusCode() != 204) {
+      throw new RuntimeException(
+          "Hourly tally sync failed with status code: "
+              + response.getStatusCode()
+              + ", response body: "
+              + response.getBody().asString());
+    }
+
+    Log.info("Hourly tally endpoint called successfully for org: %s", orgId);
+  }
+
+  public String generateRandomOrgId() {
+    return String.format("%d", Math.abs(UUID.randomUUID().hashCode()) % 100000000);
```

Comment:
There is already an utility to generate random values: RandomUtils.generateRandom in the swatch-test-framework project.